File: tests/patterns/python/dots_method_chaining.py
def foo():

    # ERROR: match
    o.foo().m().h().bar().z()

    # ERROR: match
    o.foo().bar()

    # this one does not contain the bar()
    o.foo().m().h().z()

    # ERROR: match $O can match o.before()
    o.before().foo().m().h().bar().z()

File: tests/patterns/cpp/dots_method_chaining.cpp
void test() {

  //ERROR: match
  int f = this.foo().m().h().bar().z();

  //ERROR: match
  int f = this.foo().bar();

  int f = this.foo().m().h().z();

  //ERROR: match $O can match o.before()
  int f = this.before().foo().m().h().bar().z();
}